ledger/store/src/transaction/deployment.rs

```diff
@@ -228,12 +241,27 @@ pub trait DeploymentStorage<N: Network>: Clone + Send + Sync {
             Some(program) => cow_to_cloned!(program),
             None => bail!("Failed to locate program '{program_id}' for transaction '{transaction_id}'"),
         };
+        // Check if the program and edition are in the old ID map.
+        let in_v1_id_map = self.id_map_v1().contains_key_confirmed(transaction_id)?;
 
         atomic_batch_scope!(self, {
             // Remove the program ID.
-            self.id_map().remove(transaction_id)?;
-            // Remove the edition.
-            self.edition_map().remove(&program_id)?;
+            if in_v1_id_map {
+                self.id_map_v1().remove(transaction_id)?;
+            } else {
+                self.id_map_v2().remove(transaction_id)?;
+            }
```

Comment:
perhaps we could just call remove on both maps? we're already going to need an atomic write, and this would avoid a separate lookup